Makefile

```diff
@@ -19,50 +26,48 @@ $(error "error Unexpected OS; Only Linux or Darwin supported.")
 endif
 endif
 
-# Environment
-SHELL := /bin/bash
-BASE_PATH := $(shell pwd)
-PUBLIC_PATH := $(BASE_PATH)/public
-THEMES_PATH := $(BASE_PATH)/themes
-STATIC_PATH := $(BASE_PATH)/static
-THEME_NAME := $(shell basename $(HUGO_THEME))
-THEME_PATH := $(THEMES_PATH)/$(THEME_NAME)
-HUGO_PATH := $(BASE_PATH)/.hugo
-HUGO_URL = https://github.com/spf13/hugo/releases/download/v$(HUGO_VERSION)/$(HUGO_TAR_FILE_NAME)
-HUGO_NAME := hugo_$(HUGO_VERSION)_$(ARCH)_$(OS)
-HUGO_URL_NAME := hugo_$(HUGO_VERSION)_$(URL_ARCH)-$(URL_OS)
-
 # Tools
-CURL = curl -L
-HUGO = $(HUGO_PATH)/hugo
-MKDIR = mkdir -p
-GIT = git
+CURL := curl -L
+HUGO := $(HUGO_PATH)/hugo
+MKDIR := mkdir -p
+JS_PACKAGE_MANAGER := yarn
 UNCOMPRESS := tar -zxf
 
 # Rules
 all: build
 
-init:
-	@if [ "$(HUGO_THEME)" == "" ]; then \
-		echo "ERROR! Please set the env variable 'HUGO_THEME' (http://mcuadros.github.io/autohugo/documentation/working-with-autohugo/)"; \
-		exit 1; \
-	fi;
+# Ensures the Hugo binary existance
+$(HUGO): $(HUGO_TAR_PATH)
+	$(UNCOMPRESS) $(HUGO_TAR_PATH) --directory=$(HUGO_PATH);
+
+# Downloads the hugo binary
+$(HUGO_TAR_PATH):
+	echo "Downloading $(HUGO_URL)"
+	$(MKDIR) $(HUGO_PATH)
+	$(CURL) $(HUGO_URL) -o $(HUGO_TAR_PATH)
 
-dependencies: init
-	@if [[ ! -f $(HUGO) ]]; then \
-		$(MKDIR) $(HUGO_PATH); \
-		cd $(HUGO_PATH); \
-		echo "Downloading $(HUGO_URL)"; \
-		$(CURL) $(HUGO_URL) -o $(HUGO_TAR_FILE_NAME); \
-		$(UNCOMPRESS) $(HUGO_TAR_FILE_NAME); \
-	fi;
+# Ensures hugo dependencies
+hugo-dependencies: $(HUGO_TAR_PATH) $(HUGO)
 
-build: dependencies
-	$(HUGO) -t $(THEME_NAME) --baseURL $(BASE_URL)
+# Prepares yarn
+js-dependencies:
+	$(JS_PACKAGE_MANAGER) install --force
+	$(JS_PACKAGE_MANAGER) build
+
+# Prepares project dependencies
+project-dependencies: hugo-dependencies js-dependencies
+
+# Builds hugo
+hugo-build:
+ifeq ($(ENVIRONMENT),staging)
+	$(HUGO) --baseURL=$(BASE_URL) --theme=$(HUGO_THEME_NAME) --buildDrafts --buildFuture
```

Comment:
Do upcoming blog posts need to be treated as drafts? Why not just regular published posts that we deploy to staging environment?